server/channels/api4/custom_profile_attributes_test.go

```diff
@@ -0,0 +1,247 @@
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	th.RemovePermissionFromRole(model.PermissionViewMembers.Id, model.SystemUserRoleId)
+	defer func() {
+		th.AddPermissionToRole(model.PermissionViewMembers.Id, model.SystemUserRoleId)
+	}()
+
+	field := &model.PropertyField{
+		Name: model.NewId(),
+		Type: model.PropertyFieldTypeText,
+	}
+	createdField, appErr := th.App.CreateCPAField(field)
+	require.Nil(t, appErr)
+	require.NotNil(t, createdField)
+
+	values := map[string]string{}
+	values[createdField.ID] = "Field Value"
+	_, err := th.Client.PatchCPAValues(context.Background(), values)
+	require.NoError(t, err)
+
+	t.Run("any team member should be able to list values", func(t *testing.T) {
+		th.LoginBasic2()
+		values, resp, err := th.Client.ListCPAValues(context.Background(), th.BasicUser.Id)
+		CheckOKStatus(t, resp)
+		require.NoError(t, err)
+		require.NotEmpty(t, values)
+		require.Len(t, values, 1)
+	})
+
+	t.Run("non team member should NOT be able to list values", func(t *testing.T) {
+		resp, err := th.SystemAdminClient.RemoveTeamMember(context.Background(), th.BasicTeam.Id, th.BasicUser2.Id)
+		CheckOKStatus(t, resp)
+		require.NoError(t, err)
+
+		_, resp, err = th.Client.ListCPAValues(context.Background(), th.BasicUser.Id)
+		CheckForbiddenStatus(t, resp)
+		require.Error(t, err)
+	})
+}
+
+func TestPatchCPAValues(t *testing.T) {
+	os.Setenv("MM_FEATUREFLAGS_CUSTOMPROFILEATTRIBUTES", "true")
+	defer os.Unsetenv("MM_FEATUREFLAGS_CUSTOMPROFILEATTRIBUTES")
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	field := &model.PropertyField{
+		Name: model.NewId(),
+		Type: model.PropertyFieldTypeText,
+	}
+	createdField, appErr := th.App.CreateCPAField(field)
+	require.Nil(t, appErr)
+	require.NotNil(t, createdField)
+
+	t.Run("any team member should be able to create their own values", func(t *testing.T) {
+		values := map[string]string{}
+		values[createdField.ID] = "Field Value"
+		resp, err := th.Client.PatchCPAValues(context.Background(), values)
+		CheckOKStatus(t, resp)
+		require.NoError(t, err)
+
+		values, resp, err = th.Client.ListCPAValues(context.Background(), th.BasicUser.Id)
+		CheckOKStatus(t, resp)
+		require.NoError(t, err)
+		require.NotEmpty(t, values)
+		require.Len(t, values, 1)
+	})
+
+	t.Run("any team member should be able to patch their own values", func(t *testing.T) {
+		values, resp, err := th.Client.ListCPAValues(context.Background(), th.BasicUser.Id)
+		CheckOKStatus(t, resp)
+		require.NoError(t, err)
+		require.NotEmpty(t, values)
+		require.Len(t, values, 1)
+
+		values[createdField.ID] = "Updated Field Value"
+		resp, err = th.Client.PatchCPAValues(context.Background(), values)
+		CheckOKStatus(t, resp)
+		require.NoError(t, err)
+
+		values, resp, err = th.Client.ListCPAValues(context.Background(), th.BasicUser.Id)
+		CheckOKStatus(t, resp)
+		require.NoError(t, err)
+		require.Equal(t, "Updated Field Value", values[createdField.ID])
+	})
```

Comment:
Not sure what the difference is between these two tests. The first one says `create`, but it's also patching.

Reply:
The same endpoint is used to perform multiple actions, in this case, if a field doesn't exist it will create it, and later the same endpoint call will apply a patch to the existing field. [This is the related logic](https://github.com/mattermost/mattermost/pull/29662/files#diff-74f3c860c5673d7da3357ad9454a53f2019e57ec7faeb378241994bad6e325e4R217-R235)